linkcheck/strformat.py

```diff
@@ -54,10 +56,10 @@ def unicode_safe (s, encoding=i18n.default_encoding, errors='replace'):
     @rtype: unicode
     """
     assert s is not None, "argument to unicode_safe was None"
-    if isinstance(s, unicode):
+    if isinstance(s, str_text):
         # s is already unicode, nothing to do
         return s
-    return unicode(str(s), encoding, errors)
+    return str_text(bytes(s), encoding, errors)
```

Comment:
This feels wrong to me.  Previously, `unicode_safe(42)` would return `u'42'`.  Now it will return a string with 42 NUL bytes.